### ISSUE TITLE
External links do not work when used as a Banners secondary action

### ISSUE THREAD

**Issue by cargix1** [Shopify/polaris#236]
# Issue summary

External links do not work when used as a Banners secondary action. 

## Expected behavior

I'd expect to be able to use the `external` param to open a new link to our help desk. 

## Actual behavior

The page attempts to be loaded within the EASDKs iframe.

## Steps to reproduce the problem

1. Use the following code:

```javascript
<Banner
  title="Enable Carrier Calculated Shipping"
  status="warning"
  action={{
    content: 'Enable Rates',
    onAction: () => console.log('rate enabled')
  }}
  secondaryAction={{
    content: "Alternative options",
    url: 'https://support.shopify.com',
    external: true
  }}
>
Hello World
</Banner>
```

2. Try clicking "alternative options"  and notice a new window does not open.

## Specifications

- Polaris version: 1.8.3
- React version: 15.6.1
- Browser: Chrome (latest stable)
- Device: Macbook Pro
- Operating System: High Sierra

---

⚓️ We’re not accepting pull requests at this time<br />
🗒 This repo is for reporting issues and feature requests only

**Comment by chloerice** [comment]
Hey @cargix1, thanks for bringing this to our attention! I'm working on a [fix for external links in Page action group items](https://github.com/Shopify/polaris/issues/203) right now, so I'll add a fix for this in that PR as well 😀

**Comment by dansundy** [comment]
I really need this functionality as well. @cargix1, did you happen to find a workaround?

**Comment by m20io** [comment]
@dansundy I want my external links to be open in a new tab or window, so for me this very very dirty workaround works:
```
action={{ content: 'External Action', onAction: () => { window.open('https://example.com/') } }}
```

**Comment by dansundy** [comment]
Thanks @m20io! 

I found something else that seems to work. I'm using `useLinkComponent` to convert everything into react routes. The `action` prop for the Banner will pass through an `external` argument that you can use to set a target but the `secondaryAction` doesn't. To handle that scenario I just did a manual check to see if it was an external url. Here's my `useLinkComponent` function where `base_url` is my app's url:

```
useLinkComponent(({ url, children, external, ...props }) => {
  if (external || (url.substring(0, 8) === 'https://' && url.substring(0, base_url.length) !== base_url)) {
    return (
      <a href={url} {...props} target="_blank">
        {children}
      </a>
    )
  }

  return (
    <Link to={url} {...props}>
      {children}
    </Link>
  )
}
```

**Comment by chloerice** [comment]
Nice ✨@dansundy!

**Comment by solonaarmstrong** [comment]
@dansundy @m20io @cargix1 I added external link capability to secondaryActions and it will be released in the next version update to Polaris v2.0.

**Comment by ry5n** [comment]
Polaris 2.0 is officially out of beta and this feature is now available 🎉 

Thanks everyone who suggested workarounds and contributed. 💜
